// react-instantsearch-hooks-web/dist/cjs/widgets/Breadcrumb.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Breadcrumb = Breadcrumb;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactInstantsearchHooks = require("react-instantsearch-hooks");

var _Breadcrumb = require("../ui/Breadcrumb");

var _excluded = ["attributes", "rootPath", "separator", "transformItems"];

function Breadcrumb(_ref) {
  var attributes = _ref.attributes,
      rootPath = _ref.rootPath,
      separator = _ref.separator,
      transformItems = _ref.transformItems,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);

  var _useBreadcrumb = (0, _reactInstantsearchHooks.useBreadcrumb)({
    attributes: attributes,
    rootPath: rootPath,
    transformItems: transformItems
  }, {
    $$widgetType: 'ais.breadcrumb'
  }),
      canRefine = _useBreadcrumb.canRefine,
      createURL = _useBreadcrumb.createURL,
      items = _useBreadcrumb.items,
      refine = _useBreadcrumb.refine;

  var uiProps = {
    items: items,
    hasItems: canRefine,
    createURL: createURL,
    onNavigate: refine,
    translations: {
      root: 'Home'
    }
  };
  return /*#__PURE__*/_react.default.createElement(_Breadcrumb.Breadcrumb, (0, _extends2.default)({}, props, uiProps));
}
