// react-instantsearch-hooks-web/dist/cjs/widgets/ClearRefinements.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ClearRefinements = ClearRefinements;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactInstantsearchHooks = require("react-instantsearch-hooks");

var _ClearRefinements = require("../ui/ClearRefinements");

var _excluded = ["includedAttributes", "excludedAttributes", "transformItems"];

function ClearRefinements(_ref) {
  var includedAttributes = _ref.includedAttributes,
      excludedAttributes = _ref.excludedAttributes,
      transformItems = _ref.transformItems,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);

  var _useClearRefinements = (0, _reactInstantsearchHooks.useClearRefinements)({
    includedAttributes: includedAttributes,
    excludedAttributes: excludedAttributes,
    transformItems: transformItems
  }, {
    $$widgetType: 'ais.clearRefinements'
  }),
      canRefine = _useClearRefinements.canRefine,
      refine = _useClearRefinements.refine;

  var uiProps = {
    onClick: refine,
    disabled: !canRefine,
    translations: {
      resetLabel: 'Clear refinements'
    }
  };
  return /*#__PURE__*/_react.default.createElement(_ClearRefinements.ClearRefinements, (0, _extends2.default)({}, props, uiProps));
}
