// react-instantsearch-hooks-web/dist/cjs/widgets/HierarchicalMenu.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HierarchicalMenu = HierarchicalMenu;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactInstantsearchHooks = require("react-instantsearch-hooks");

var _HierarchicalMenu = require("../ui/HierarchicalMenu");

var _excluded = ["attributes", "limit", "rootPath", "separator", "showMore", "showMoreLimit", "showParentLevel", "sortBy", "transformItems"];

function HierarchicalMenu(_ref) {
  var attributes = _ref.attributes,
      limit = _ref.limit,
      rootPath = _ref.rootPath,
      separator = _ref.separator,
      showMore = _ref.showMore,
      showMoreLimit = _ref.showMoreLimit,
      showParentLevel = _ref.showParentLevel,
      sortBy = _ref.sortBy,
      transformItems = _ref.transformItems,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);

  var _useHierarchicalMenu = (0, _reactInstantsearchHooks.useHierarchicalMenu)({
    attributes: attributes,
    limit: limit,
    rootPath: rootPath,
    separator: separator,
    showMore: showMore,
    showMoreLimit: showMoreLimit,
    showParentLevel: showParentLevel,
    sortBy: sortBy,
    transformItems: transformItems
  }, {
    $$widgetType: 'ais.hierarchicalMenu'
  }),
      items = _useHierarchicalMenu.items,
      canRefine = _useHierarchicalMenu.canRefine,
      canToggleShowMore = _useHierarchicalMenu.canToggleShowMore,
      createURL = _useHierarchicalMenu.createURL,
      isShowingMore = _useHierarchicalMenu.isShowingMore,
      refine = _useHierarchicalMenu.refine,
      toggleShowMore = _useHierarchicalMenu.toggleShowMore;

  var uiProps = {
    items: items,
    hasItems: canRefine,
    createURL: createURL,
    onNavigate: refine,
    canToggleShowMore: canToggleShowMore,
    onToggleShowMore: toggleShowMore,
    isShowingMore: isShowingMore
  };
  return /*#__PURE__*/_react.default.createElement(_HierarchicalMenu.HierarchicalMenu, (0, _extends2.default)({}, props, uiProps, {
    showMore: showMore
  }));
}
