// react-instantsearch-hooks-web/dist/cjs/widgets/RangeInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RangeInput = RangeInput;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _reactInstantsearchHooks = require("react-instantsearch-hooks");

var _RangeInput = require("../ui/RangeInput");

var _excluded = ["attribute", "min", "max", "precision"];

function RangeInput(_ref) {
  var attribute = _ref.attribute,
      min = _ref.min,
      max = _ref.max,
      precision = _ref.precision,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);

  var _useRange = (0, _reactInstantsearchHooks.useRange)({
    attribute: attribute,
    min: min,
    max: max,
    precision: precision
  }, {
    $$widgetType: 'ais.rangeInput'
  }),
      range = _useRange.range,
      start = _useRange.start,
      canRefine = _useRange.canRefine,
      refine = _useRange.refine;

  var step = 1 / Math.pow(10, precision || 0);
  var uiProps = {
    range: range,
    start: start,
    step: step,
    disabled: !canRefine,
    onSubmit: refine,
    translations: {
      separator: 'to',
      submit: 'Go'
    }
  };
  return /*#__PURE__*/_react.default.createElement(_RangeInput.RangeInput, (0, _extends2.default)({}, props, uiProps));
}
