// @medusajs/medusa-js/dist/resources/admin/sales-channels.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var base_1 = __importDefault(require("../base"));
var qs_1 = __importDefault(require("qs"));
var AdminSalesChannelsResource = /** @class */ (function (_super) {
    __extends(AdminSalesChannelsResource, _super);
    function AdminSalesChannelsResource() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /** retrieve a sales channel
     * @experimental This feature is under development and may change in the future.
     * To use this feature please enable featureflag `sales_channels` in your medusa backend project.
     * @description gets a sales channel
     * @returns a medusa sales channel
     */
    AdminSalesChannelsResource.prototype.retrieve = function (salesChannelId, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels/".concat(salesChannelId);
        return this.client.request("GET", path, undefined, {}, customHeaders);
    };
    /* *
     * Create a medusa sales channel
     * @returns the created channel
     */
    AdminSalesChannelsResource.prototype.create = function (payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels";
        return this.client.request("POST", path, payload, {}, customHeaders);
    };
    /** update a sales channel
     * @experimental This feature is under development and may change in the future.
     * To use this feature please enable featureflag `sales_channels` in your medusa backend project.
     * @description updates a sales channel
     * @returns the updated medusa sales channel
     */
    AdminSalesChannelsResource.prototype.update = function (salesChannelId, payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels/".concat(salesChannelId);
        return this.client.request("POST", path, payload, {}, customHeaders);
    };
    /**
     * Retrieve a list of sales channels
     * @experimental This feature is under development and may change in the future.
     * To use this feature please enable featureflag `sales_channels` in your medusa backend project.
     * @description Retrieve a list of sales channels
     * @returns the list of sales channel as well as the pagination properties
     */
    AdminSalesChannelsResource.prototype.list = function (query, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels";
        if (query) {
            var queryString = qs_1.default.stringify(query);
            path += "?".concat(queryString);
        }
        return this.client.request("GET", path, undefined, {}, customHeaders);
    };
    /**
     * Delete a sales channel
     * @experimental This feature is under development and may change in the future.
     * To use this feature please enable featureflag `sales_channels` in your medusa backend project.
     * @description gets a sales channel
     * @returns an deletion result
     */
    AdminSalesChannelsResource.prototype.delete = function (salesChannelId, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels/".concat(salesChannelId);
        return this.client.request("DELETE", path, undefined, {}, customHeaders);
    };
    /**
     * Remove products from a sales channel
     * @experimental This feature is under development and may change in the future.
     * To use this feature please enable featureflag `sales_channels` in your medusa backend project.
     * @description Remove products from a sales channel
     * @returns a medusa sales channel
     */
    AdminSalesChannelsResource.prototype.removeProducts = function (salesChannelId, payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels/".concat(salesChannelId, "/products/batch");
        return this.client.request("DELETE", path, payload, {}, customHeaders);
    };
    /**
     * Add products to a sales channel
     * @experimental This feature is under development and may change in the future.
     * To use this feature please enable featureflag `sales_channels` in your medusa backend project.
     * @description Add products to a sales channel
     * @returns a medusa sales channel
     */
    AdminSalesChannelsResource.prototype.addProducts = function (salesChannelId, payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/sales-channels/".concat(salesChannelId, "/products/batch");
        return this.client.request("POST", path, payload, {}, customHeaders);
    };
    return AdminSalesChannelsResource;
}(base_1.default));
exports.default = AdminSalesChannelsResource;
