// @medusajs/medusa-js/dist/resources/admin/collections.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var qs_1 = __importDefault(require("qs"));
var base_1 = __importDefault(require("../base"));
var AdminCollectionsResource = /** @class */ (function (_super) {
    __extends(AdminCollectionsResource, _super);
    function AdminCollectionsResource() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * @description Creates a collection.
     * @param payload
     * @param customHeaders
     * @returns Created collection.
     */
    AdminCollectionsResource.prototype.create = function (payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections";
        return this.client.request("POST", path, payload, {}, customHeaders);
    };
    /**
     * @description Updates a collection
     * @param id id of the collection to update.
     * @param payload update to apply to collection.
     * @param customHeaders
     * @returns the updated collection.
     */
    AdminCollectionsResource.prototype.update = function (id, payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections/".concat(id);
        return this.client.request("POST", path, payload, {}, customHeaders);
    };
    /**
     * @description deletes a collection
     * @param id id of collection to delete.
     * @param customHeaders
     * @returns Deleted response
     */
    AdminCollectionsResource.prototype.delete = function (id, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections/".concat(id);
        return this.client.request("DELETE", path, undefined, {}, customHeaders);
    };
    /**
     * @description get a collection
     * @param id id of the collection to retrieve.
     * @param customHeaders
     * @returns the collection with the given id
     */
    AdminCollectionsResource.prototype.retrieve = function (id, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections/".concat(id);
        return this.client.request("GET", path, undefined, {}, customHeaders);
    };
    /**
     * @description Lists collections matching a query
     * @param query Query for searching collections
     * @param customHeaders
     * @returns a list of collections matching the query.
     */
    AdminCollectionsResource.prototype.list = function (query, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections";
        if (query) {
            var queryString = qs_1.default.stringify(query);
            path = "/admin/collections?".concat(queryString);
        }
        return this.client.request("GET", path, undefined, {}, customHeaders);
    };
    /**
     * @description Updates products associated with a Product Collection
     * @param id the id of the Collection
     * @param payload - an object which contains an array of Product IDs to add to the Product Collection
     * @param customHeaders
     */
    AdminCollectionsResource.prototype.addProducts = function (id, payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections/".concat(id, "/products/batch");
        return this.client.request("POST", path, payload, {}, customHeaders);
    };
    /**
     * @description Removes products associated with a Product Collection
     * @param id - the id of the Collection
     * @param payload - an object which contains an array of Product IDs to add to the Product Collection
     * @param customHeaders
     */
    AdminCollectionsResource.prototype.removeProducts = function (id, payload, customHeaders) {
        if (customHeaders === void 0) { customHeaders = {}; }
        var path = "/admin/collections/".concat(id, "/products/batch");
        return this.client.request("DELETE", path, payload, {}, customHeaders);
    };
    return AdminCollectionsResource;
}(base_1.default));
exports.default = AdminCollectionsResource;
