// react-instantsearch-hooks-web/dist/es/widgets/Breadcrumb.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["attributes", "rootPath", "separator", "transformItems"];
import React from 'react';
import { useBreadcrumb } from 'react-instantsearch-hooks';
import { Breadcrumb as BreadcrumbUiComponent } from "../ui/Breadcrumb.js";
export function Breadcrumb(_ref) {
  var attributes = _ref.attributes,
      rootPath = _ref.rootPath,
      separator = _ref.separator,
      transformItems = _ref.transformItems,
      props = _objectWithoutProperties(_ref, _excluded);

  var _useBreadcrumb = useBreadcrumb({
    attributes: attributes,
    rootPath: rootPath,
    transformItems: transformItems
  }, {
    $$widgetType: 'ais.breadcrumb'
  }),
      canRefine = _useBreadcrumb.canRefine,
      createURL = _useBreadcrumb.createURL,
      items = _useBreadcrumb.items,
      refine = _useBreadcrumb.refine;

  var uiProps = {
    items: items,
    hasItems: canRefine,
    createURL: createURL,
    onNavigate: refine,
    translations: {
      root: 'Home'
    }
  };
  return /*#__PURE__*/React.createElement(BreadcrumbUiComponent, _extends({}, props, uiProps));
}
