// react-instantsearch-hooks-web/dist/es/widgets/RangeInput.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["attribute", "min", "max", "precision"];
import React from 'react';
import { useRange } from 'react-instantsearch-hooks';
import { RangeInput as RangeInputUiComponent } from "../ui/RangeInput.js";
export function RangeInput(_ref) {
  var attribute = _ref.attribute,
      min = _ref.min,
      max = _ref.max,
      precision = _ref.precision,
      props = _objectWithoutProperties(_ref, _excluded);

  var _useRange = useRange({
    attribute: attribute,
    min: min,
    max: max,
    precision: precision
  }, {
    $$widgetType: 'ais.rangeInput'
  }),
      range = _useRange.range,
      start = _useRange.start,
      canRefine = _useRange.canRefine,
      refine = _useRange.refine;

  var step = 1 / Math.pow(10, precision || 0);
  var uiProps = {
    range: range,
    start: start,
    step: step,
    disabled: !canRefine,
    onSubmit: refine,
    translations: {
      separator: 'to',
      submit: 'Go'
    }
  };
  return /*#__PURE__*/React.createElement(RangeInputUiComponent, _extends({}, props, uiProps));
}
